### DISCUSSION TITLE
Upgrade Database from v21.9.1 to v24.8.0

### DISCUSSION THREAD

**Discussion by uktricky** [librenms/docker#461] [Q&A]
Hi, I've been running Librenms for sometime but never updated it!!! I've been using the [jarischaefer] docker image originally but looking to use the now provided librenms one.   I've attempt to mysqldump the original v21.9.1 and restore this into v24.8.0 and when I run "./lnms migrate" I get the following:

 INFO  Running migrations.

  2021_07_28_102443_plugins_add_version_and_settings ................................................................ 1,962ms DONE
  2021_09_07_094310_create_push_subscriptions_table ..................................................................... 2ms FAIL

In Connection.php line 829:

  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'push_subscriptions' already exists (Connection: mysql, SQL: cre
  ate table `push_subscriptions` (`id` bigint unsigned not null auto_increment primary key, `subscribable_type` varchar(255) not
  null, `subscribable_id` bigint unsigned not null, `endpoint` varchar(500) not null, `public_key` varchar(255) null, `auth_token
  ` varchar(255) null, `content_encoding` varchar(255) null, `description` varchar(255) null, `created_at` timestamp null, `updat
  ed_at` timestamp null) default character set utf8mb4 collate 'utf8mb4_unicode_ci')

In Connection.php line 587:

  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'push_subscriptions' already exists

librenms:/opt/librenms#

Can anyone help/advise my steps needed to move this onto a new version of librenms on a new device...

**Comment by laf** [comment]
If you are running the database and librenms install in two separate docker containers then don't dump the DB, just roll out the new librenms container and run the lnms migrate command.

**Reply by uktricky** [reply]
> If you are running the database and librenms install in two separate docker containers then don't dump the DB, just roll out the new librenms container and run the lnms migrate command.

Thanks for a quick response - So remove the db section from the example/compose file and use my existing MariaDB?  Or would I create the db using the example before I create the database and restore my 21.9.1 database into this and then install using the compose file?  I am moving the whole setup to a new device hence me wanting to run a separate version before I blow away the original older one.

**Reply by laf** [reply]
If you are moving to a new device then you should be able to run the new librenms docker image, dump the database on your old one and import to the new platform then run lnms migrate. If that's what you have done already then it looks like you're migrations are out of sync for some reason.

**Reply by uktricky** [reply]
> If you are moving to a new device then you should be able to run the new librenms docker image, dump the database on your old one and import to the new platform then run lnms migrate. If that's what you have done already then it looks like you're migrations are out of sync for some reason.

That's exactly what I'd done other than I had added a couple of test devices to prove it worked.  I expected those would have been overwritten on the restore?  I've also noticed that I am very low on disk space on the testing I am doing so that might also be a reason.  

I'll blow it away tomorrow and start a scratch doing the database restore before trying to use the app.  

Thanks again.

**Reply by uktricky** [reply]
Guessing the database in use from jarischaefer has been amended in some way.  I've reset everything today and ensured I've plenty disk space remaining and get exactly the same errors as my original post.  I've just used the example compose.yml file to spin up LibreNMS, then attempted to restore the full database from my old LibreNMS, then the lnms migrate...
